test/parallel/test-fs-readfile-tostring-fail.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const kStringMaxLength = process.binding('buffer').kStringMaxLength;
+
+common.refreshTmpDir();
+
+const file = path.join(common.tmpDir, 'toobig.txt');
+const stream = fs.createWriteStream(file, {
+  flags: 'a'
+});
+
+const size = kStringMaxLength / 200;
+const a = new Buffer(size).fill('a');
+
+for (var i = 0; i < 201; i++) {
+  stream.write(a);
+}
+
+stream.end();
+stream.on('finish', common.mustCall(function() {
+  // make sure that the toString does not throw an error
+  fs.readFile(file, 'utf8', common.mustCall(function(err, buf) {
+    assert.ok(err instanceof Error);
+    assert.strictEqual('toString failed', err.message);
+  }));
+}));
+
+process.on('exit', function() {
+  fs.unlinkSync(file);
+});
+
+// To make sure we don't leave a very large file
+// on test machines in the event this test fails.
+process.on('uncaughtException', function(err) {
```

Comment:
may want to add `process.on('SIGINT'` to capture any `^C`. would be the same as here, except run `process.exit()` at the end. optional, non-blocker.